indicatorStatusIcon.js

```diff
@@ -403,18 +403,31 @@ class IndicatorStatusIcon extends BaseStatusIcon {
         }
 
         if (event.get_button() === Clutter.BUTTON_SECONDARY) {
-            this.menu.toggle();
+            if (this.menu.numMenuItems) {
+                this.menu.toggle();
+            } else {
+                // update menu if there's no entries
+                this._updateMenu();
+                // wait for menu update
+                this._waitForDoubleClick().catch(logError);
+            }
```

Comment:
I'm unsure here, shouldn't be:

```suggestion
            // Try to update the menu if there are no entries.
            if (!this.menu.numMenuItems)
                this._updateMenu();

            this.menu.toggle();
```

Reply:
Thank you.

It seems that `this.menu.toggle()` is called inside `this._waitForDoubleClick()` so does not need to be called here.